### ISSUE TITLE
Using CPU with Docker

### ISSUE THREAD

**Issue by thisismydesign** [keras-team/keras#8667]
Hi,

TLDR;
Could you please clarify in the README how to run Keras in docker using CPU? Maybe also create a separate Dockerfile that doesn't install GPU related stuff making the size and initial setup time much more pleasant.

Issue:

I found [docker/README](https://github.com/fchollet/keras/tree/d5a242e9f6c17a60bbbc7bd114b34ba59dfcca29/docker) misleading as it mentions "GPU support" as a separate item even though that [seems to be the default](https://github.com/fchollet/keras/blob/d5a242e9f6c17a60bbbc7bd114b34ba59dfcca29/docker/Makefile#L7). I think that's problematic because the [platform support](https://github.com/NVIDIA/nvidia-docker/wiki/Frequently-Asked-Questions#platform-support) for `nvidia-docker` is pretty narrow.

Currently I'm trying to make Keras work via Docker from MacOS with no success. Upon running the container via `make notebook` I create a notebook where I try to follow the steps from the [README](https://github.com/fchollet/keras#getting-started-30-seconds-to-keras).

`from keras.models import Sequential`

```
Using TensorFlow backend.
---------------------------------------------------------------------------
ImportError                               Traceback (most recent call last)
/opt/conda/lib/python3.5/site-packages/tensorflow/python/pywrap_tensorflow.py in <module>()
     57 
---> 58   from tensorflow.python.pywrap_tensorflow_internal import *
     59   from tensorflow.python.pywrap_tensorflow_internal import __version__

/opt/conda/lib/python3.5/site-packages/tensorflow/python/pywrap_tensorflow_internal.py in <module>()
     27             return _mod
---> 28     _pywrap_tensorflow_internal = swig_import_helper()
     29     del swig_import_helper

/opt/conda/lib/python3.5/site-packages/tensorflow/python/pywrap_tensorflow_internal.py in swig_import_helper()
     23             try:
---> 24                 _mod = imp.load_module('_pywrap_tensorflow_internal', fp, pathname, description)
     25             finally:

/opt/conda/lib/python3.5/imp.py in load_module(name, file, filename, details)
    242         else:
--> 243             return load_dynamic(name, filename, file)
    244     elif type_ == PKG_DIRECTORY:

/opt/conda/lib/python3.5/imp.py in load_dynamic(name, path, file)
    342             name=name, loader=loader, origin=path)
--> 343         return _load(spec)
    344 

ImportError: libcuda.so.1: cannot open shared object file: No such file or directory

During handling of the above exception, another exception occurred:

ImportError                               Traceback (most recent call last)
<ipython-input-2-9c5e0a19b646> in <module>()
----> 1 from keras.models import Sequential

/src/keras/__init__.py in <module>()
      1 from __future__ import absolute_import
      2 
----> 3 from . import utils
      4 from . import activations
      5 from . import applications

/src/keras/utils/__init__.py in <module>()
      4 from . import data_utils
      5 from . import io_utils
----> 6 from . import conv_utils
      7 
      8 # Globally-importable utils.

/src/keras/utils/conv_utils.py in <module>()
      1 from six.moves import range
      2 import numpy as np
----> 3 from .. import backend as K
      4 
      5 

/src/keras/backend/__init__.py in <module>()
     81 elif _BACKEND == 'tensorflow':
     82     sys.stderr.write('Using TensorFlow backend.\n')
---> 83     from .tensorflow_backend import *
     84 else:
     85     raise ValueError('Unknown backend: ' + str(_BACKEND))

/src/keras/backend/tensorflow_backend.py in <module>()
----> 1 import tensorflow as tf
      2 from tensorflow.python.training import moving_averages
      3 from tensorflow.python.ops import tensor_array_ops
      4 from tensorflow.python.ops import control_flow_ops
      5 from tensorflow.python.ops import functional_ops

/opt/conda/lib/python3.5/site-packages/tensorflow/__init__.py in <module>()
     22 
     23 # pylint: disable=wildcard-import
---> 24 from tensorflow.python import *
     25 # pylint: enable=wildcard-import
     26 

/opt/conda/lib/python3.5/site-packages/tensorflow/python/__init__.py in <module>()
     47 import numpy as np
     48 
---> 49 from tensorflow.python import pywrap_tensorflow
     50 
     51 # Protocol buffers

/opt/conda/lib/python3.5/site-packages/tensorflow/python/pywrap_tensorflow.py in <module>()
     70 for some common reasons and solutions.  Include the entire stack trace
     71 above this error message when asking for help.""" % traceback.format_exc()
---> 72   raise ImportError(msg)
     73 
     74 # pylint: enable=wildcard-import,g-import-not-at-top,unused-import,line-too-long

ImportError: Traceback (most recent call last):
  File "/opt/conda/lib/python3.5/site-packages/tensorflow/python/pywrap_tensorflow.py", line 58, in <module>
    from tensorflow.python.pywrap_tensorflow_internal import *
  File "/opt/conda/lib/python3.5/site-packages/tensorflow/python/pywrap_tensorflow_internal.py", line 28, in <module>
    _pywrap_tensorflow_internal = swig_import_helper()
  File "/opt/conda/lib/python3.5/site-packages/tensorflow/python/pywrap_tensorflow_internal.py", line 24, in swig_import_helper
    _mod = imp.load_module('_pywrap_tensorflow_internal', fp, pathname, description)
  File "/opt/conda/lib/python3.5/imp.py", line 243, in load_module
    return load_dynamic(name, filename, file)
  File "/opt/conda/lib/python3.5/imp.py", line 343, in load_dynamic
    return _load(spec)
ImportError: libcuda.so.1: cannot open shared object file: No such file or directory


Failed to load the native TensorFlow runtime.

See https://www.tensorflow.org/install/install_sources#common_installation_problems

for some common reasons and solutions.  Include the entire stack trace
above this error message when asking for help.

```

Which then led me here: https://github.com/tensorflow/tensorflow/issues/10642

It looks like (as with the make command) GPU is used by default. I tried:
- setting `device = cpu` in [theanorc](https://github.com/fchollet/keras/blob/d5a242e9f6c17a60bbbc7bd114b34ba59dfcca29/docker/theanorc)
- Installing `tensorflow` instead of `tensorflow-gpu` in the [Dockerfile](https://github.com/fchollet/keras/blob/d5a242e9f6c17a60bbbc7bd114b34ba59dfcca29/docker/Dockerfile#L33)

No success.

**Comment by kaczmarj** [comment]
In the Dockerfile, can you try replacing the [nvidia base image](https://github.com/keras-team/keras/blob/d5a242e9f6c17a60bbbc7bd114b34ba59dfcca29/docker/Dockerfile#L3) with something like `debian:stretch` in addition to installing `tensorflow` instead of `tensorflow-gpu`?

```Dockerfile
FROM debian:stretch
```

**Comment by PieterjanCriel** [comment]
The `Debian:stretch` image will work with some additional installs. The `bzip2` package, required by miniconda, is missing. In order to fix this add the following in the Dockerfile

```Dockerfile
FROM debian:stretch

ENV DEBIAN_FRONTEND noninteractive
RUN apt-get update -qq \
 && apt-get install --no-install-recommends -y \
    bzip2 \
 && apt-get clean \
 && rm -rf /var/lib/apt/lists/*
```

remove packages associated with GPU support and use `tensorflow` instead of `tensorflow-gpu`

```Dockerfile
RUN conda install -y python=${python_version} && \
    pip install --upgrade pip && \
    pip install tensorflow && \
    conda install Pillow scikit-learn notebook pandas matplotlib mkl nose pyyaml six h5py && \
    pip install sklearn_pandas && \
    git clone git://github.com/keras-team/keras.git /src && pip install -e /src[tests] && \
    pip install git+git://github.com/keras-team/keras.git && \
    conda clean -yt
```

**Comment by kaczmarj** [comment]
It might be useful to start with the [continuumio/miniconda3](https://hub.docker.com/r/continuumio/miniconda3/) base image.

**Comment by kouml** [comment]
According to Zen of Python(Explicit is better than implicit), I think it is better for adding cpu version of Dockerfile in this repository.
